src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,157 @@
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineMethod;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(PredefinedCodeRefactoringProviderNames.InlineMethod)), Shared]
+    [Export(typeof(CSharpInlineMethodRefactoringProvider))]
+    internal sealed class CSharpInlineMethodRefactoringProvider : AbstractInlineMethodRefactoringProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpInlineMethodRefactoringProvider() : base(CSharpSyntaxFacts.Instance)
+        {
+        }
+
+        protected override async Task<SyntaxNode?> GetInvocationExpressionSyntaxNodeAsync(CodeRefactoringContext context)
+        {
+            var syntaxNode = await context.TryGetRelevantNodeAsync<InvocationExpressionSyntax>().ConfigureAwait(false);
+            return syntaxNode;
+        }
+
+        private static bool ShouldStatementBeInlined(StatementSyntax statementSyntax)
+            => statementSyntax is ReturnStatementSyntax || statementSyntax is ExpressionStatementSyntax;
+
+        protected override bool IsMethodContainsOneStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
+            {
+                var blockSyntaxNode = declarationSyntax.Body;
+                // 1. If it is an ordinary method with block
+                if (blockSyntaxNode != null)
+                {
+                    var blockStatements = blockSyntaxNode.Statements;
+                    return blockStatements.Count == 1 && ShouldStatementBeInlined(blockStatements[0]);
+                }
+                else
+                {
+                    // 2. If it is an Arrow Expression
+                    var arrowExpressionNodes = declarationSyntax
+                        .DescendantNodes().Where(node => node.IsKind(SyntaxKind.ArrowExpressionClause)).ToImmutableArray();
+                    return arrowExpressionNodes.Length == 1;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel, SyntaxNode argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode is ArgumentSyntax argumentSyntax
+                ? argumentSyntax.DetermineParameter(semanticModel, allowParams: true, cancellationToken)
+                : null;
+
+        protected override SyntaxNode GenerateLiteralExpression(ITypeSymbol typeSymbol, object? value)
+            => ExpressionGenerator.GenerateExpression(typeSymbol, value, canUseFieldReference: false);
```

Comment:
why `canUseFieldReference: false`.  seems like we'd prefer the field-ref option here.

Reply:
Found SyntaxGenerator has the LiteralExpression Method. Change it to use that instead.